### DISCUSSION TITLE
Authenticating Username and Password by removing hard coaded values.

### DISCUSSION THREAD

**Discussion by Anirudh-Suri** [ktbyers/netmiko#2871] [Q&A] [answered]
I want to write a script to find the Network devices in a network. But how can we authenticate UN and PASSWORD with out taking the hardcoded values in the script suppose in dictionary?

**Comment by ktbyers** [comment, marked as answer]
You can use `input` to prompt for credentials, you can use OS environment variables and read those, you can use some sort of encrypted data store (that you then provide the key for in some way).
